boa_engine/src/vm/call_frame/mod.rs

```diff
@@ -33,6 +33,9 @@ pub struct CallFrame {
     // When an async generator is resumed, the generator object is needed
     // to fulfill the steps 4.e-j in [AsyncGeneratorStart](https://tc39.es/ecma262/#sec-asyncgeneratorstart).
     pub(crate) async_generator: Option<JsObject>,
+
+    // Iterators and their `[[Done]]` flags that must be closed when an abrupt completion is thrown.
+    pub(crate) iterators: Vec<(JsObject, bool)>,
```

Comment:
```suggestion
    pub(crate) iterators: ThinVec<(JsObject, bool)>,
```
I think it would be good to make this a thin vec to minimize call frame size on functions that don't use iterators